src/analysis/calculus/deriv.lean

```diff
@@ -121,10 +121,20 @@ lemma has_fderiv_within_at_iff_has_deriv_within_at {f' : 𝕜 →L[𝕜] F} :
   has_fderiv_within_at f f' s x ↔ has_deriv_within_at f (f' 1) s x :=
 by simp [has_deriv_within_at, has_deriv_at_filter, has_fderiv_within_at]
 
+lemma has_deriv_within_at_iff_has_fderiv_within_at {f' : F} :
```

Comment:
The existing iff lemmas around here seem to go in the other direction -- `fderiv` on the left, `deriv` on the right. Is there a reason this one goes the other way? Naively, the LHS looks simpler to me.

Reply:
All these lemmas will most of the time be used from left to right, in the following way: you have a simple expression, but you may want to transform it into something more complicated to which you will apply an existing theorem. The other direction is essentially not used, as there is no reason why you would stumble on something of this specific form in a proof. This is why all these lemmas are written with the simple expression on the left, and without a simp attribute.

Reply:
I see. I think the naming is a little confusing -- it sounds like the symmetric version of the lemma right before. But I don't have a better suggestion. Maybe it would help to have doc strings on both directions that explain the difference? It would show up if someone tried to use one when they needed the other.

In any case, I don't think it's a very big deal.

Reply:
I have added docstrings, hopefully they clarify things at least a little bit.